brave/browser-laptop
# Extension data no longer removed in clear cookies…

… action. Called method is now cookie specific.

Fixes Issue #13098

## Submitter Checklist:


## Test Plan:

Test plan specified in issue:

1. Configure Brave to clear all site cookies on close in about:preferences#security.
2. Enable and sign-into the LastPass extension
3. Restart Brave
4. Lastpass should not prompt for credentials

## Reviewer Checklist:


Tests



Previously, there was one very general method being called when "clear all site cookies" was enabled called clearStorageData(). This deleted: appcache, cookies, filesystem, indexdb, local storage,shadercache, websql, and serviceworkers. This was way too aggressive. 

Per https://github.com/brave/muon/blob/master/atom/browser/api/atom_api_session.cc#L84

We can see that specific storage types can be specified, the only one we should need to touch for this option is "cookies"

In the filtering module, specific clearCache() and clearHistory() methods exist. I created an pseudo alias of the clearStorageData() method called clearCookies() that is called instead. bound to the clearStorageData() call is `{'storages': ['cookies']}` which specifies the cookies storage type.

See: https://github.com/electron/electron/blob/master/docs/api/session.md#sesclearstoragedataoptions-callback
